sql/core/src/test/scala/org/apache/spark/sql/execution/command/DDLSuite.scala

```diff
@@ -1376,32 +1398,67 @@ abstract class DDLSuite extends QueryTest with DDLSuiteBase {
     }
   }
 
-  test("SPARK-51747: Data source cached plan should respect options") {
-    withTable("t") {
-      spark.sql("CREATE TABLE t(a string, b string) USING CSV".stripMargin)
-      spark.sql("INSERT INTO TABLE t VALUES ('a;b', 'c')")
-      spark.sql("INSERT INTO TABLE t VALUES ('hello; world', 'test')")
+  test("SPARK-51747: Data source cached plan respects options if ignore conf disabled") {
+    Seq("true", "false").foreach { ignoreOption =>
+      withSQLConf(SQLConf.READ_FILE_SOURCE_TABLE_CACHE_IGNORE_OPTIONS.key -> ignoreOption) {
+        withNamespace("ns") {
+          withTable("t") {
+            spark.sql("CREATE TABLE t(a string, b string) USING CSV".stripMargin)
+            spark.sql("INSERT INTO TABLE t VALUES ('a;b', 'c')")
+            spark.sql("INSERT INTO TABLE t VALUES ('hello; world', 'test')")
 
-      // check initial contents of table
-      checkAnswer(spark.table("t"), Row("a;b", "c") :: Row("hello; world", "test") :: Nil)
+            // check initial contents of table
+            checkAnswer(spark.table("t"), Row("a;b", "c") :: Row("hello; world", "test") :: Nil)
 
-      // no option
-      checkAnswer(
-        spark.sql("SELECT * FROM t"),
-        Row("a;b", "c") :: Row("hello; world", "test") :: Nil
-      )
+            val shouldIgnoreOption = ignoreOption == "true"
```

Comment:
i think its a bit useless.  How about just doing 'ignoreOption' in the seq, and then putting 

SQLConf.READ_FILE_SOURCE_TABLE_CACHE_IGNORE_OPTIONS.key -> ignoreOption.toString

then it make the rest of the code shorter (without need for 'should' variable)